Add discontiguous kwarg to make_tensor

heitorschueroff opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #51807 Implemented torch.linalg.multi_dot
* #52464 Add torch.linalg to generated annotated_args for test_overrides
* **#51985 Add discontiguous kwarg to make_tensor**

Differential Revision: [D26375733](https://our.internmc.facebook.com/intern/diff/D26375733)

heitorschueroff committed:
Add discontiguous kwarg to make_tensor

[ghstack-poisoned]

mruberry reviewed (commented):
This looks great. Would you add a test to test_testing.py, like `test_make_tensor_discontiguous`, that accepts different input sizes (empty tensor, scalar tensor, single element 1D, single element 2D (1x1), 1D, 2D, 3D) and verifies that this returns a tensor that has the same values but with a discontiguous innermost dimension. It can run on bool, long, float and cfloat.

What does this do for tensors with only a single element?

heitorschueroff committed:
Update on "Add discontiguous kwarg to make_tensor"




[ghstack-poisoned]

heitorschueroff committed:
Update on "Add discontiguous kwarg to make_tensor"




[ghstack-poisoned]

heitorschueroff committed:
Update on "Add discontiguous kwarg to make_tensor"




[ghstack-poisoned]

heitorschueroff committed:
Update on "Add discontiguous kwarg to make_tensor"


Differential Revision: [D26375733](https://our.internmc.facebook.com/intern/diff/D26375733)

[ghstack-poisoned]

mruberry commented on test/test_testing.py:
```diff
@@ -435,6 +435,17 @@ def test_isclose_atol_rtol_greater_than_zero(self, device, dtype):
+            self.assertTrue(res.is_contiguous())
+        # The following sizes should be discontiguous
+        for size in ((2,), (2, 3), (2, 1, 3)):
+            res = make_tensor(size, device, dtype, low=None, high=None, discontiguous=True)
```
Add a size check for both cases

mruberry reviewed (approved):
Cool! Just add a size check to the test.

heitorschueroff committed:
Update on "Add discontiguous kwarg to make_tensor"


Differential Revision: [D26375733](https://our.internmc.facebook.com/intern/diff/D26375733)

[ghstack-poisoned]

heitorschueroff committed:
Update on "Add discontiguous kwarg to make_tensor"


Differential Revision: [D26375733](https://our.internmc.facebook.com/intern/diff/D26375733)

[ghstack-poisoned]

heitorschueroff committed:
Update on "Add discontiguous kwarg to make_tensor"


Differential Revision: [D26375733](https://our.internmc.facebook.com/intern/diff/D26375733)

[ghstack-poisoned]

heitorschueroff committed:
Update on "Add discontiguous kwarg to make_tensor"


Differential Revision: [D26375733](https://our.internmc.facebook.com/intern/diff/D26375733)

[ghstack-poisoned]

heitorschueroff committed:
Update on "Add discontiguous kwarg to make_tensor"


Differential Revision: [D26375733](https://our.internmc.facebook.com/intern/diff/D26375733)

[ghstack-poisoned]

heitorschueroff committed:
Update on "Add discontiguous kwarg to make_tensor"


Differential Revision: [D26375733](https://our.internmc.facebook.com/intern/diff/D26375733)

[ghstack-poisoned]

heitorschueroff committed:
Update on "Add discontiguous kwarg to make_tensor"


Differential Revision: [D26375733](https://our.internmc.facebook.com/intern/diff/D26375733)

[ghstack-poisoned]

heitorschueroff committed:
Update on "Add discontiguous kwarg to make_tensor"


Differential Revision: [D26375733](https://our.internmc.facebook.com/intern/diff/D26375733)

[ghstack-poisoned]
